feature.go

```diff
@@ -0,0 +1,237 @@
+package lago
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type FeatureRequest struct {
+	client *Client
+}
+
+type PrivilegeConfig struct {
+	SelectOptions []string `json:"select_options,omitempty"`
+}
+
+type Privilege struct {
+	Code      string          `json:"code"`
+	Name      string          `json:"name,omitempty"`
+	ValueType string          `json:"value_type"` // TODO: add enum
+	Config    PrivilegeConfig `json:"config"`
+}
+
+type Feature struct {
+	Name        string      `json:"name,omitempty"`
+	Code        string      `json:"code"`
+	Description string      `json:"description,omitempty"`
+	Privileges  []Privilege `json:"privileges,omitempty"`
+	CreatedAt   time.Time   `json:"created_at,omitempty"`
+}
+
+type FeatureResult struct {
+	Feature  *Feature  `json:"feature,omitempty"`
+	Features []Feature `json:"features,omitempty"`
+	Meta     Metadata  `json:"meta,omitempty"`
+}
+
+type FeatureListInput struct {
+	PerPage int `json:"per_page,omitempty,string"`
+	Page    int `json:"page,omitempty,string"`
+}
+
+type FeatureParams struct {
+	FeatureInput *FeatureInput `json:"feature,omitempty"`
+}
+
+type FeatureInput struct {
+	Name        string           `json:"name,omitempty"`
+	Code        string           `json:"code,omitempty"`
+	Description string           `json:"description,omitempty"`
+	Privileges  []PrivilegeInput `json:"privileges,omitempty"`
+}
+
+type ConfigInput struct {
+	SelectOptions []string `json:"select_options,omitempty"`
+}
+
+type PrivilegeInput struct {
+	Code      string      `json:"code,omitempty"`
```

Comment:
Code cannot be empty:

```suggestion
	Code      string      `json:"code"`
```